src/tasks/trainerUtils.py

```diff
@@ -113,5 +191,329 @@ def generate_trainer_name(raw_trainer_name, pokemon1_level):
+                        print("This trainer is currently not supported or is not in a recognized area:", args[0].strip("'"), areaName)                                
+        return trainers
+
+def get_support_trainers_data(file_path, area_name, support_name, zoneID):
+    """
+    The purpose of this function is to get the support trainers data
+    """
+    temp_support_IDs = []
+
+    for support in [support_name]:
+        temp_support_IDs = parse_random_teams(file_path, f"{area_name.lower()}_{support}_100", 3, None)
+    if temp_support_IDs == []:
+        temp_support_IDs = parse_random_teams(file_path, f"ev_{area_name.lower()}_rivalfight_get_rival_team", 3, None)
+    if temp_support_IDs == []:
+        temp_support_IDs = parse_random_teams(file_path, f"{area_name.lower()}_rival_support", 3, None)
+    if temp_support_IDs == [] and support_name != None:
+        for support in [support_name]:
+            if support == "male":
+                temp_support_IDs = parse_random_teams(file_path, f"ev_r207_func_17", 3, None)
+                break
+            elif support == "female":
+                temp_support_IDs = parse_random_teams(file_path, f"ev_r207_func_20", 3, None)
+                break
+    if temp_support_IDs == []:
+        temp_support_IDs = parse_random_teams(file_path, f"ev_r212ar0101_func_19", 3, None)
+    if temp_support_IDs == []:
+        print("Support Trainers still needs more work", area_name, zoneID)
+    trainers = []
+    for ID in temp_support_IDs:
+        trainer = get_trainer_data(None, zoneID, int(ID))
+        trainer["format"] = "Multi"
+        trainer["link"] = "Support"
+        trainers.append(trainer)
+    return trainers
+
+def get_single_trainer(zoneID, ID, temp_IDs, name):
+    trainer = get_trainer_data(None, zoneID, int(ID))
+    starters = ["piplup", "turtwig" ,"chimchar"]
+    master_trainer_types = ["fire", "water", "electric", "grass", "ice", "fighting", "poison", "ground", "flying", "psychic", "bug", "rock", "ghost", "dragon", "dark", "steel", "fairy"]
+    if name in starters:
+        trainer["name"] = f"{trainer['name']} {name} Team {str(temp_IDs.index(ID) + 1)}"
+    elif name in master_trainer_types:
+        trainer["name"] = f"{name.capitalize()} Master Trainer {trainer['name']}"
+    elif name == "Celebi":
+        trainer["name"] = f"{trainer['name']}"
+    elif len(temp_IDs) > 1:
+        trainer["name"] = f"{trainer['name']} Team {str(temp_IDs.index(ID) + 1)}"
+    else:
+        trainer["name"] = f"{trainer['name']}"
+    
+    trainer["format"] = "Single"
+    trainer["link"] = ""
     
-get_trainer_data_from_place_datas()
+    return trainer
+
+def get_multi_trainers(trainerID1, trainerID2, zoneID, format):
+    '''
+    format can either be "Multi" or "Double" depending on which you choose
+    '''
+    trainers = []
+    trainerID2_used = False
+    for trainerID in [trainerID1, trainerID2]:
+        if trainerID:
```

Comment:
Deep nesting. You can resolve one layer by checking the negative of `if trainerID` instead.